### PR TITLE
[SWE-58] Update Documentation

### PR DESCRIPTION
**Pull request by datderek**
Task ID: [SWE-58](https://linear.app/swe-dem-boys/issue/SWE-58/update-documentation)
## PR Description
Update documentation to make it presentable for resume purposes:
- Comment complicated code for readability
- Mark uncompleted sections with TODO comments
- Update API Documentation
- Update README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,3 +69,7 @@ npm run start
 ```
 
 ## Run Backend
+
+# 📚 Documentation
+For documentation on how to utilize our API please visit this 
+[link.](https://documenter.getpostman.com/view/31017970/2s9YkuYy3k)
```
Comment (jackfales) [inline README.md:75]:
```suggestion
For documentation on how to utilize our API please see our Postman API [docs.](https://documenter.getpostman.com/view/31017970/2s9YkuYy3k)
```

File: README.md [hunk 2]
```diff
@@ -1,16 +1,45 @@
...
+
+🥕**Virtual Pantry**: Keep track of your ingredients with our virtual pantry! 
+Easily see what you already have for a given recipe and replenish ingredients 
+that you're running low on.
+
+🤖**AI-Powered Recommendations**: Receive personalized dish recommendations based 
+on your previous interests so you'll never have to scroll too far to find 
+something you'll love!
+
+# 🛠️ Local Development
```
Comment (jackfales) [inline README.md:36]: This section is fine as is

File: app/[user]/page.tsx [hunk 3]
```diff
@@ -3,7 +3,9 @@ import { headers } from "next/headers";
 import Image from 'next/image';
 import Navbar from "../components/Navbar";
 import { EditProfileAndFollowButton } from "../components/ClientUtilityFunctions";
-// TODO These imports may not be needed when we implement a database
+/* TODO: Remove the imports below and static user data located at `./data` and
```
Comment (jackfales) [inline app/[user]/page.tsx:6]: Make a linear task where this TODO will be completed and change it to TODO(SWE-XX) where XX is the task id.
```suggestion
/* TODO(SWE-XX): Remove the imports below and static user data located at `./data` and
```

File: app/[user]/page.tsx [hunk 4]
```diff
@@ -12,7 +14,9 @@ export const metadata = {
 }
 
 const profileDirectory: string = path.join(process.cwd(), 'data');
-
+/* TODO: Update generateStaticParams function to query the graphDB for all
```
Comment (jackfales) [inline app/[user]/page.tsx:17]: Same

File: app/[user]/page.tsx [hunk 5]
```diff
@@ -27,6 +31,9 @@ export async function generateStaticParams() {
   })
 }
 
+/* TODO: Update getProfileData function to query the graphDB for the user data
```
Comment (jackfales) [inline app/[user]/page.tsx:34]: Same

File: app/[user]/page.tsx [hunk 6]
```diff
@@ -41,16 +48,18 @@ export function getProfileData(id: string): any {
 
 
 export default async function Profile({ params }: { params: {user: string}}) {
+  /* TODO: Replace this authorization check with a check at the API gateway */
```
Comment (jackfales) [inline app/[user]/page.tsx:51]: Same

File: app/[user]/page.tsx [hunk 7]
```diff
@@ -41,16 +48,18 @@ export function getProfileData(id: string): any {
...
   const req = {
     headers: {
       cookie: headers().get("cookie"),
     },
   };
 
-  // Passes client-side credentials to server via cookies
   const { Auth } = withSSRContext({ req });
 
+  // TODO: Need to redirect to `/login` similar to `./dashboard/page.tsx`
```
Comment (jackfales) [inline app/[user]/page.tsx:61]: Same

File: app/dashboard/page.tsx [hunk 8]
```diff
@@ -14,18 +16,18 @@ export const metadata = {
   title: 'Dashboard'
 }
 
+//TODO: Remove these constants as these were for the static data
```
Comment (jackfales) [inline app/dashboard/page.tsx:19]:
```suggestion
// TODO: Remove these constants as these were for the static data
```

File: pages/create-account.tsx [hunk 9]
```diff
@@ -69,19 +70,21 @@ export default function CreateAccount() {
...
     setErrorMessages({"confirmPassword" : errors["confirmPassword"],
                       "email" : errors["email"],
                       "firstName" : errors["firstName"],
                       "lastName" : errors["lastName"],
                       "password" : errors["password"],
                       "username" : errors["username"]
                       });
-
-    // Check if the form has any errors
+
```
Comment (jackfales) [inline pages/create-account.tsx:80]:
```suggestion

```

File: pages/forgot-password.tsx [hunk 10]
```diff
@@ -15,14 +15,20 @@ function ForgotPassword() {
...
     } else {
         setErrorMessage("")
     }
 
-    // Confirmation code sent to user's email
+    /* TODO: Separate backend from frontend. The following code should be 
+     * refactored to an Route Handler. 
+     * Similar to what is done in `./app/components/CreatePostModal.tsx`
+     */        
+    // Sends an reset password to the email associated with the username
```
Comment (jackfales) [inline pages/forgot-password.tsx:31]:
```suggestion
    // Sends an reset password to the email associated with the username
    //
    // TODO: Separate backend from frontend. The following code should be 
    // refactored to an Route Handler. 
    // Similar to what is done in `./app/components/CreatePostModal.tsx`    
```

**Review by jackfales — CHANGES_REQUESTED** [review]
Update TODOs with linear task IDs for easy tracking

File: app/components/EditProfileForm.tsx
Comment (sammyoto) [inline app/components/EditProfileForm.tsx:1]: SWE-29 just got pulled. May want to pull main to this branch to get updates.

**Review by sammyoto — CHANGES_REQUESTED** [review]
Overall looks good. Just one comment needs fixing.

**Review by sammyoto — APPROVED** [review]
Looks good to me.

**Review by jackfales — APPROVED** [review]
Nice work

**Pull request merged**